examples/federated-database-query-limit/main.tf

```diff
@@ -0,0 +1,71 @@
+resource "mongodbatlas_cluster" "atlas_cluster_1" {
+  project_id                  = var.project_id
+  provider_name               = var.provider_name
+  name                        = var.atlas_cluster_name_1
+  backing_provider_name       = var.backing_provider_name
+  provider_region_name        = var.provider_region_name
+  provider_instance_size_name = var.provider_instance_size_name
+}
+
+
+resource "mongodbatlas_cluster" "atlas_cluster_2" {
+  project_id                  = var.project_id
+  provider_name               = var.provider_name
+  name                        = var.atlas_cluster_name_2
+  backing_provider_name       = var.backing_provider_name
+  provider_region_name        = var.provider_region_name
+  provider_instance_size_name = var.provider_instance_size_name
+}
+
+resource "mongodbatlas_federated_database_instance" "test-instance" {
+  project_id = var.project_id
+  name       = var.federated_instance_name
+  aws {
+    role_id        = ""
+    test_s3_bucket = ""
+  }
```

Comment:
Shoun't we use variables here?

Reply:
`aws` is a required field in the federated instance resource. @themantissa we will need to make this an optional field first in this case. Please confirm.
cc @andreaangiolillo

Reply:
So let's take a step back first and cover why both @andreaangiolillo and I are commenting on this. We should not have empty parameters like this.  We've never had it yet so we need to figure out why and if they are needed.  

To Zuhair's point - he's correct.  There are two AWS areas.   I thought this referred to the stores at first glance but he's correct, it doesn't.  It looks like a holding place for when this supports more than one cloud provider.  Right now it's just AWS and the docs note it's only "Required if specifying cloudProviderConfig".  I would think you'd only need to specify this when it moves beyond a single cloud provider.    Really this should be within a  cloudProviderConfig block so perhaps we need to revisit the PR there (https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1163/files#diff-ebd5e08c7fa8e4f0fee0f3b1c8c687c52fd98ce77434a1053e591ee3952d7902R69).  Furthermore, I don't see how a user would need to provide this- wouldn't that be a MDB S3 bucket this federation is stored in?  Do we store it in the users bucket and if so this would likely need data from cloud provider access - which means we need a better example of that.   Can we perhaps reach out to the Data Federation team to get their help there?

Reply:
reached out to @bafMdb to help comment on each of the above items shortly. also agree with point on not having empty parameters in our examples going forward, at the very least what we can do is call these out as user inputted strings in variables.tf file and then reference, i.e. role_id = var.role_id, test_s3_bucket = var.test_s3_bucket, etc.

Reply:
Pulling from the public docs here:
`{
"cloudProviderConfig": {
"aws": {
"roleId": "string",
"testS3Bucket": "string"
}
},
"dataProcessRegion": {
"cloudProvider": "AWS",
"region": "SYDNEY_AUS"
},
"name": "string",
"storage": {
"databases": [],
"stores": []
}
}`

Cloud provider config should only be necessary if you are querying object storage, you should not need to use anything there if you are just created a federated database instance to query a cluster.

Separately data process region allows you select the cloud provider where data gets processed and the region.

Both of these should map over into a world where we support Azure or GCP.

Reply:
Thanks. I think we should make the `aws` optional in the [federated database instance](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/0c6184da0a713c14109c5435b868a01ed353bb69/mongodbatlas/resource_mongodbatlas_federated_database_instance.go#L53).

Reply:
No concerns, thanks both @maastha and @andreaangiolillo

Reply:
@andreaangiolillo @Zuhairahmed  I do have some concerns - I'm assuming this is in a `cloudProviderConfig` block because eventually it would have other cloud providers beyond AWS, e.g. Azure, GCP, for their object storage.   We can certainly solve this just to make each, in turn, optional in Terraform and if we want to move faster that's a way we can choose to go. But I think this makes for harder to read HCL code - it means a dangling cloud block that even we had a hard time figuring out why it's there.  If we end up with 3 cloud providers we'll have 3 of these.   Really part of the issue here is naming.   I think it would have been better for the API parameter to have a name like cloudProviderObjectStorageConfig but only @baf509 and team can explain why the shorter but harder to figure out name.   So I think we should note in the docs that aws{} is required if using object storage and at least separate it in Terraform in some way like:

cloudProviderConfig {
   aws {
      role_id        = ""
      test_s3_bucket = ""
   }
 //to make space for when we eventually have:
   azure {
    //required variables if using object storage
   }
   gcp {
    //required variables if using object storage
   }
}

Reply:
It's good feedback. Current approach should be fine in short term, but as soon as we get support Azure or Google Cloud here it will get confusing as you point out @themantissa. I also think this is a matter of when, not if so suggest we invest the time this time around to make clear from the start. 

@andreaangiolillo any concerns adding back in cloudProviderConfig in doc fix per above in mongodbatlas_federated_database_instance? If you are also aligned would also then to make minor commits in this PR to adjust examples accordingly here as well.

Reply:
@Zuhairahmed @themantissa updated. Thanks!